### DISCUSSION TITLE
[PlutoPkg] Support for Dev Packages / Non-Registered packages

### DISCUSSION THREAD

**Discussion by kfardanesh** [fonsp/Pluto.jl#1302] [Pluto development]
### Problem
I upgraded to the latest release of Pluto (0.15.x) and found that a bunch of my notebooks broke because I was using packages not registered in the registry. Unfortunately I had to revert to using the `Pkg.activate()`

I haven't taken a look yet at how PlutoPkg detects which packages to add to the notebook environment, however it might be a good idea to support dev packages and packages not registered.

### Proposal
A proposed solution would be to "extend" `using <package name>` with a comment to point to the appropriate local path or url of the package as such:
`using <package name> # /path/to/local/package`
`using <package name> # https://github.com/non-registered/package`

Might need to differentiate between `Pkg.add` and `Pkg.develop`

I am open to any and all ideas around this issue

**Comment by swadey** [comment]
I'm in the same boat.  A bunch of notebooks I have depend on PlutoDataTable which isn't registered.  Is there a good workaround?

**Reply by disberd** [reply]
Direct support for this in the way suggsted by @kfardanesh would be idea.
In the meantime you can add a Local registry using something like https://github.com/GunnarFarneback/LocalRegistry.jl to add a local registry where you keep an up-to date list of packages that you want to use.
PlutoPkg gets the packages from all registries that you ave added on your Julia instance as mentioned [here](https://github.com/fonsp/Pluto.jl/wiki/%F0%9F%8E%81-Package-management#custom-registries)

I am currently using this approach to have the benefit of the new package manager with my own custom packges that are not registered in General (don't know if it works for Pkg.develop)

**Comment by kfardanesh** [comment]
Linking similarly related issue around `Pkg.develop` and PlutoPkg co-existing peacefully: #1304
